### PR TITLE
Remove python2 specifics from docs and unix tests

### PR DESCRIPTION
**Pull request by TimSpeelman**
As discussed in #485, I replaced the outdated `python2` with `python` througout the docs. Also, I removed the python2 specific test from the Unix tests, otherwise this may throw a confusing error when the user `pip installs` to python3 but the tests decide to run v2.

### CODE REVIEW

**Comment by qstokkink** [comment]
ok to test

File: run_all_tests_unix.sh [hunk 1]
```diff
@@ -8,13 +8,7 @@ test_files="$(grep ^[^#] test_classes_list.txt)"
...
 tty -s && tput bold
 
-# 2.1 Check for python interpreter version
-if [ -x "$(command -v python2)" ]; then
-    interpreter="python2"
-else
-    echo "No python2 command found! Will use the default python command."
-    interpreter="python"
-fi
+interpreter="python"
```
Comment (qstokkink) [inline run_all_tests_unix.sh]: We can't change this right now, our PR testing suite on Jenkins depends on this being Python 2.

**Comment by TimSpeelman** [comment]
Ik don't understand why the checks fail, when trying to open the Flake8 details I get a 403.

**Comment by qstokkink** [comment]
Our flake8 test assumes all PRs modify at least some code. You can ignore the failure.

**Pull request merged**